exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -118,13 +133,34 @@ public boolean apply(@Nullable FileStatus status) {
       }
     }));
 
-    return create(nonDirectories, null, selectionRoot);
+    final FileSelection fileSel = create(nonDirectories, null, selectionRoot);
```

Comment:
Agreed that it's existing create() call. It seems fine to keep it as is.  

This new FileSelection is special, since it has excluded directories by going through the list of file status recursively. In that sense, it's hasDirectories = false, checkedForDirectories=true, and the list of files are already available while going through the file status list. If a FileSelection always has to call containsDirectory() or getFile() eventually, then it might be better to store them when they are available. But I guess it's minor thing to consider.